docs/src/main/asciidoc/security-openid-connect-multitenancy.adoc

```diff
@@ -268,14 +267,19 @@ public class CustomTenantResolver implements TenantResolver {
 
-However, using a static tenant resolution (configuring tenants in `application.properties` and resolving them with `TenantResolver`) prevents testing the endpoint with `Dev Services for Keycloak` since `Dev Services for Keycloak` has no knowledge of how the requests will be mapped to individual tenants and can not dynamically provide tenant-specific `quarkus.oidc.<tenant-id>.auth-server-url` values and therefore using `%prod` prefixes with the tenant-specific URLs in `application.properties` will not work in tests or devmode.
 
 [NOTE]
 ====
-When a current tenant represents an OIDC `web-app` application, the current `io.vertx.ext.web.RoutingContext` will contain a `tenant-id` attribute by the time the custom tenant resolver has been called for all the requests completing the code authentication flow and the already authenticated requests, when either a tenant specific state or session cookie already exists.
-Therefore, when working with multiple OpenID Connect Providers, you only need a path specific check to resolve a tenant id if the `RoutingContext` does not have the `tenant-id` attribute set, for example:
+When a current tenant represents an OIDC `web-app` application, the current `io.vertx.ext.web.RoutingContext` contains a `tenant-id` attribute by the time the custom tenant resolver has been called for all the requests completing the code authentication flow and the already authenticated requests, when either a tenant-specific state or session cookie already exists.
+Therefore, when working with multiple OIDC providers, you only need a path-specific check to resolve a tenant ID if the `RoutingContext` does not have the `tenant-id` attribute set, for example:
+
+// When dealing with an OIDC `web-app` application as the current tenant, the process involves specific attributes and contexts. In this scenario, the `io.vertx.ext.web.RoutingContext` of the current tenant contains a `tenant-id` attribute. This inclusion occurs by the time the custom tenant resolver is called. It applies to all requests that complete the code authentication flow and to already authenticated requests. This authentication is recognized when a tenant-specific state or session cookie already exists.
+//
+// Therefore, when working with multiple OIDC providers, resolving a tenant ID requires careful consideration. Specifically, a path-specific check becomes necessary only if the `RoutingContext` lacks the `tenant-id` attribute. For example:
```

Comment:
@sberyozkin I revised lines 277-278 to break the original complex sentences into shorter, clearer sentences. Each new sentence introduces a specific step or concept, enhancing readability and comprehension, in line with effective technical writing practices. 

However, I want to make sure I have preserved the intended meaning. If you approve this revision. I will remove lines 277-278 and uncomment 280-282.

Reply:
@rolfedh Thanks for giving it try, I'm not sure though rewriting the 1st sentence does not change the meaning, let me try to compare. The current sentence: 

```
When a current tenant represents an OIDC `web-app` application, the current `io.vertx.ext.web.RoutingContext` contains a `tenant-id` attribute by the time the custom tenant resolver has been called for all the requests completing the code authentication flow and the already authenticated requests, when either a tenant-specific state or session cookie already exists.
```

With `When a current tenant represents an OIDC web-app application` I'm trying to say that a configuration for the OIDC `web-app` exists that matches the current tenant. It feels that `When dealing with an OIDC web-app application as the current tenant` does not quite work.

With `the current io.vertx.ext.web.RoutingContext contains a `tenant-id` attribute` I try to say  `the current request's  io.vertx.ext.web.RoutingContext` so `io.vertx.ext.web.RoutingContext` of the current tenant` does not quite work either.

Also, IMHO, `the process involves specific attributes and contexts` addition is OK but reads a little bit technical.

Thanks

Reply:
To expedite this PR, I've removed my comment/suggestion:
```
// When dealing with an OIDC `web-app` application as the current tenant, the process involves specific attributes and contexts.
// In this scenario, the `io.vertx.ext.web.RoutingContext` of the current tenant contains a `tenant-id` attribute.
// This inclusion occurs by the time the custom tenant resolver is called.
// It applies to all requests that complete the code authentication flow and to already authenticated requests.
// This authentication is recognized when a tenant-specific state or session cookie already exists.
//
// Therefore, when working with multiple OIDC providers, resolving a tenant ID requires careful consideration.
// Specifically, a path-specific check becomes necessary only if the `RoutingContext` lacks the `tenant-id` attribute.
// For example:
```

Reply:
If you have the time, please suggest revisions to make following passage easier to understand. 
```
When a current tenant represents an OIDC `web-app` application, the current `io.vertx.ext.web.RoutingContext` contains a `tenant-id` attribute by the time the custom tenant resolver has been called for all the requests completing the code authentication flow and the already authenticated requests, when either a tenant-specific state or session cookie already exists.
Therefore, when working with multiple OIDC providers, you only need a path-specific check to resolve a tenant ID if the `RoutingContext` does not have the `tenant-id` attribute set, for example:
```

Reply:
@sberyozkin, for clarity, I am marking this conversation as resolved. I have opened [a separate request](https://github.com/quarkusio/quarkus/pull/37584#discussion_r1447621226) asking @pedroigor for help rewriting one line.